File: Challenges/bigNumbers.js
var fs = require("fs");
fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    if (line !== "") {

        var line1 = "";
        var line2 = "";
        var line3 = "";
        var line4 = "";
        var line5 = "";
        var line6 = "";
        
        for (var i = 0; i <= line.length ; ++i) {
            if (line[i] === "0") {
                line1 += "-**--";
                line2 += "*--*-";
                line3 += "*--*-";
                line4 += "*--*-";
                line5 += "-**--";
                line6 += "-----";
            } else if (line[i] == "1") {
                line1 += "--*--";
                line2 += "-**--";
                line3 += "--*--";
                line4 += "--*--";
                line5 += "-***-";
                line6 += "-----";
            } else if (line[i] == "2") {
                line1 += "***--";
                line2 += "---*-";
                line3 += "-**--";
                line4 += "*----";
                line5 += "****-";
                line6 += "-----";
            } else if (line[i] == "3") {
                line1 += "***--";
                line2 += "---*-";
                line3 += "-**--";
                line4 += "---*-";
                line5 += "***--";
                line6 += "-----";
            } else if (line[i] == "4") {
                line1 += "-*---";
                line2 += "*--*-";
                line3 += "****-";
                line4 += "---*-";
                line5 += "---*-";
                line6 += "-----";
            } else if (line[i] == "5") {
                line1 += "****-";
                line2 += "*----";
                line3 += "***--";
                line4 += "---*-";
                line5 += "***--";
                line6 += "-----";
            } else if (line[i] == "6") {
                line1 += "-**--";
                line2 += "*----";
                line3 += "***--";
                line4 += "*--*-";
                line5 += "-**--";
                line6 += "-----";
            } else if (line[i] == "7") {
                line1 += "****-";
                line2 += "---*-";
                line3 += "--*--";
                line4 += "-*---";
                line5 += "-*---";
                line6 += "-----";
            } else if (line[i] == "8") {
                line1 += "-**--";
                line2 += "*--*-";
                line3 += "-**--";
                line4 += "*--*-";
                line5 += "-**--";
                line6 += "-----";
            } else if (line[i] == "9") {
                line1 += "-**--";
                line2 += "*--*-";
                line3 += "-***-";
                line4 += "---*-";
                line5 += "-**--";
                line6 += "-----";
            } else {
                
            }
        }
        console.log(line1);
        console.log(line2);
        console.log(line3);
        console.log(line4);
        console.log(line5);
        console.log(line6);
    }
});

File: Challenges/bigNumbers.py
#Codeval - Big Numbers

import sys
test_cases = open(sys.argv[1], 'r')
for test in test_cases:
	# 'test' represents the test case
	line1 = ""
	line2 = ""
	line3 = ""
	line4 = ""
	line5 = ""
	line6 = ""
	for element in test:
	# building Big Number output
		if element == "0":
			line1 += "-**--"
			line2 += "*--*-"
			line3 += "*--*-"
			line4 += "*--*-"
			line5 += "-**--"
			line6 += "-----"
		elif element == "1":
			line1 += "--*--"
			line2 += "-**--"
			line3 += "--*--"
			line4 += "--*--"
			line5 += "-***-"
			line6 += "-----"
		elif element == "2":
			line1 += "***--"
			line2 += "---*-"
			line3 += "-**--"
			line4 += "*----"
			line5 += "****-"
			line6 += "-----"
		elif element == "3":   	
			line1 += "***--"
			line2 += "---*-"
			line3 += "-**--"
			line4 += "---*-"
			line5 += "***--"
			line6 += "-----"
		elif element == "4":    	
			line1 += "-*---"
			line2 += "*--*-"
			line3 += "****-"
			line4 += "---*-"
			line5 += "---*-"
			line6 += "-----"
		elif element == "5":    	
			line1 += "****-"
			line2 += "*----"
			line3 += "***--"
			line4 += "---*-"
			line5 += "***--"
			line6 += "-----"
		elif element == "6":    	
			line1 += "-**--"
			line2 += "*----"
			line3 += "***--"
			line4 += "*--*-"
			line5 += "-**--"
			line6 += "-----"
		elif element == "7":    	
			line1 += "****-"
			line2 += "---*-"
			line3 += "--*--"
			line4 += "-*---"
			line5 += "-*---"
			line6 += "-----"
		elif element == "8":    	
			line1 += "-**--"
			line2 += "*--*-"
			line3 += "-**--"
			line4 += "*--*-"
			line5 += "-**--"
			line6 += "-----"
		elif element == "9":
			line1 += "-**--"
			line2 += "*--*-"
			line3 += "-***-"
			line4 += "---*-"
			line5 += "-**--"
			line6 += "-----"
		else:
			line1 += ""
			line2 += ""
			line3 += ""
			line4 += ""
			line5 += ""
			line6 += ""

	print line1 + '\n', line2 + '\n', line3 + '\n', line4 + '\n', line5 + '\n', line6

test_cases.close()
